opsdroid/connector/slack/connector.py

```diff
@@ -0,0 +1,375 @@
+            )
+            if config.get("socket-mode")
+            else None
+        )
+        self.bot_name = config.get("bot-name", "opsdroid")
+        self.auth_info = None
+        self.user_info = None
+        self.bot_id = None
+        self.known_users = {}
+
+        self._event_creator = SlackEventCreator(self)
+
+    async def connect(self):
+        """Connect to the chat service."""
+        _LOGGER.info(_("Connecting to Slack."))
+
+        try:
+            self.auth_info = (await self.slack_web_client.api_call("auth.test")).data
+            self.user_info = (
+                await self.slack_web_client.api_call(
+                    "users.info",
+                    http_verb="GET",
+                    params={"user": self.auth_info["user_id"]},
+                )
+            ).data
+            self.bot_id = self.user_info["user"]["profile"]["bot_id"]
+
+            if self.socket_mode_client:
+                self.socket_mode_client.socket_mode_request_listeners.append(
+                    self.socket_event_handler
+                )
+                await self.socket_mode_client.connect()
+            else:
+                self.opsdroid.web_server.web_app.router.add_post(
+                    "/connector/{}".format(self.name),
+                    self.web_event_handler,
+                )
+
+            _LOGGER.debug(_("Connected as %s."), self.bot_name)
+            _LOGGER.debug(_("Using icon %s."), self.icon_emoji)
+            _LOGGER.debug(_("Default room is %s."), self.default_target)
+            _LOGGER.info(_("Connected successfully."))
+        except SlackApiError as error:
+            _LOGGER.error(
+                _(
+                    "Unable to connect to Slack due to %s."
+                    "The Slack Connector will not be available."
+                ),
+                error,
+            )
+
+    async def disconnect(self):
+        """Disconnect from Slack. Only needed when socket_mode_client is used
+        as the Events API uses the aiohttp server"""
+
+        if self.socket_mode_client:
+            await self.socket_mode_client.disconnect()
+            await self.socket_mode_client.close()
+
+    async def listen(self):
+        """Listen for and parse new messages."""
+
+    async def event_handler(self, payload):
+        event = None
+
+        if "type" in payload:
+            if payload["type"] == "event_callback":
+                event = await self._event_creator.create_event(payload["event"], None)
+            elif payload["type"] in (
+                "block_actions",
+                "message_action",
+                "view_submission",
+                "view_closed",
+            ):
+                event = await self._event_creator.create_event(payload, None)
+            else:
+                _LOGGER.info(
+                    f"Payload: {payload['type']} is not implemented. Event wont be parsed"
+                )
```

Comment:
By default the event creator should return None if you give it an event it doesn't understand. This means that you don't have to add support for new events in more than one place.

Also if you don't have an event to parse you can return early saving you a type check later on.
```suggestion
            else:
                event = await self._event_creator.create_event(payload, None)
                if event is None:
                    _LOGGER.info(
                        f"Payload: {payload['type']} is not implemented. Event wont be parsed"
                    )
                    return
```

Reply:
the event starts already as None on line 118. So I don't see the point on calling again the event_creator.

I do agree on returning early to save type checks.